onedal/datatypes/_data_conversion.py

```diff
@@ -18,28 +18,62 @@
 
 import numpy as np
 
-from daal4py.sklearn._utils import make2d
 from onedal import _backend, _is_dpc_backend
 
-from ..utils import _is_csr
-
 
 def _apply_and_pass(func, *args, **kwargs):
     if len(args) == 1:
         return func(args[0], **kwargs)
     return tuple(map(lambda arg: func(arg, **kwargs), args))
 
 
-if _is_dpc_backend:
+def _convert_one_to_table(arg):
+    # All inputs for table conversion must be array-like or sparse, not scalars
+    return _backend.to_table(np.atleast_2d(arg) if np.isscalar(arg) else arg)
+
+
+def to_table(*args):
+    """Create oneDAL tables from scalars and/or arrays.
+
+    Note: this implementation can be used with contiguous scipy.sparse, numpy
+    arrays, sycl_usm_ndarrays, and scalars. Tables will use pointers to the
+    original array data. Scalars will be copies. Arrays may be modified in-
+    place by oneDAL during computation. This works for data located on CPU and
+    SYCL-enabled Intel GPUs. Each array may only be of a single datatype (i.e.
+    each must be homogeneous).
+
+    Parameters
+    ----------
+    *args : {scalar, numpy array, sycl_usm_ndarray, csr_matrix, or csr_array}
+        arg1, arg2... The arrays should be given as arguments.
 
-    from ..utils._dpep_helpers import dpctl_available, dpnp_available
+    Returns
+    -------
+    tables: {oneDAL homogeneous tables}
+    """
+    return _apply_and_pass(_convert_one_to_table, *args)
 
-    if dpctl_available:
-        import dpctl.tensor as dpt
 
-    if dpnp_available:
+if _is_dpc_backend:
+
+    try:
         import dpnp
 
+        def _onedal_gpu_table_to_array(table, xp=None):
+            # By default DPNP ndarray created with a copy.
+            # TODO:
+            # investigate why dpnp.array(table, copy=False) doesn't work.
+            # Work around with using dpctl.tensor.asarray.
+            if xp == dpnp:
+                return dpnp.array(dpnp.dpctl.tensor.asarray(table), copy=False)
```

Comment:
I have already asked this on #2126, but let me ask it here once again
I don't know is this dpctl module in dpnp is an official part of API? This may work now, but I am not sure for further versions of DPNP. It worth to add some comments here

Reply:
Good question https://github.com/IntelPython/dpnp/blob/master/dpnp/__init__.py#L38  It looks like they haven't figured out how to use ```os.add_dll_directory``` and are importing dpctl to get the file path and add it to the environment path. So as long as they need to find dpctl's shared objects, they will do it this way.